lib/openstudio-standards/standards/ashrae_90_1_prm/ashrae_90_1_prm.AirLoopHVAC.rb

```diff
@@ -81,36 +85,61 @@ def air_loop_hvac_apply_prm_baseline_fan_power(air_loop_hvac)
       end
     elsif system_type == 'PTAC' ||
           system_type == 'PTHP' ||
-          (system_type == 'Gas_Furnace' && !is_nmc) ||
-          (system_type == 'Electric_Furnace' && !is_nmc)
-      # Determine allowable fan power
-      allowable_power_w = 0.3
-    elsif (system_type == 'Gas_Furnace' && is_nmc) ||
-          (system_type == 'Electric_Furnace' && is_nmc)
+          system_type == 'Gas_Furnace' ||
+          system_type == 'Electric_Furnace'
+
       # Determine allowable fan power
-      allowable_power_w = 0.054
+      if !is_nmc
+        fan_efficacy_w_per_cfm = 0.3
+      else # is_nmc
+        fan_efficacy_w_per_cfm = 0.054
+      end
+
+      # Configuration is supply fan only
+      supply_fan_power_fraction = 1.0
     end
 
     supply_fan = air_loop_hvac_get_supply_fan(air_loop_hvac)
     if supply_fan.nil?
       OpenStudio.logFree(OpenStudio::Error, 'openstudio.ashrae_90_1_prm.AirLoopHVAC', "Supply not found on #{airloop.name}.")
     end
+    supply_fan_max_flow = if supply_fan.autosizedMaximumFlowRate.is_initialized
+                            supply_fan.autosizedMaximumFlowRate.get
+                          else
+                            supply_fan.maximumFlowRate.get
+                          end
 
     # Check that baseline system has the same
     # types of fans as the proposed model, if
     # not, create them. We assume that the
     # system has at least a supply fan.
     if return_fan_power_fraction > 0.0 && !air_loop_hvac.returnFan.is_initialized
       # Create return fan
-      return_fan = supply_fan.clone(model)
+      return_fan = supply_fan.clone(air_loop_hvac.model)
+      if return_fan.to_FanConstantVolume.is_initialized
+        return_fan = return_fan.to_FanConstantVolume.get
+      elsif return_fan.to_FanVariableVolume.is_initialized
+        return_fan = return_fan.to_FanVariableVolume.get
+      elsif return_fan.to_FanOnOff.is_initialized
+        return_fan = return_fan.to_FanOnOff.get
+      end
       return_fan.setName("#{air_loop_hvac.name} Return Fan")
       return_fan.addToNode(air_loop_hvac.returnAirNode.get)
+      return_fan.setMaximumFlowRate(supply_fan_max_flow)
```

Comment:
Assume that the return fan has the same maximum flow rate as the supply fan.

Reply:
Is it possible that the suply_fan would be of type FanSystemModel? Thus, would need to add check for that type here, and also in the test_appendix_g_prm.rb script.